File: 2023_manha/trabalhoFila_atendimento/Atendimento.java
import java.util.LinkedList;
import java.util.Queue;

import javax.swing.JOptionPane;  

public class Atendimento {
    public static void main(String[] args) {
        Queue<Ficha> filaNormal = new LinkedList<Ficha>();
        Queue<Ficha> filaPrioritaria = new LinkedList<Ficha>();
        int contadorFilaNormal = 1;
        int contadorFilaPrioritaria = 1000;
        int contadorAtendimentos = 3;
        String opcao = "";
        Ficha ficha;
        do {
            try {
                opcao = JOptionPane.showInputDialog(null,"1 - Ficha normal\n2 - Ficha prioritaria\n3 - Chama atendimento\n");
                switch (opcao) {
                    case "1":
                        ficha = new Ficha(contadorFilaNormal);
                        filaNormal.add(ficha);
                        contadorFilaNormal++;
                        JOptionPane.showMessageDialog(null,"Voce retirou a ficha normal ..... " + ficha.getConteudo() + " - " + ficha.getDataHora());        
                        contadorAtendimentos++;
                        break;
                    case "2":
                        ficha = new Ficha(contadorFilaPrioritaria);
                        filaPrioritaria.add(ficha);
                        contadorFilaPrioritaria++;
                        JOptionPane.showMessageDialog(null,"Voce retirou a ficha prioritaria ..... " + ficha.getConteudo() + " - " + ficha.getDataHora());                                
                        contadorAtendimentos++;
                        break;
                    case "3":  
                        //System.out.println("asdfasdf asdf as" + (contadorAtendimentos % 3)  + " ssafasfd" +contadorAtendimentos);                  
                        if (contadorAtendimentos % 3 == 1 && !filaPrioritaria.isEmpty()) {
                            ficha = filaPrioritaria.peek();                        
                            JOptionPane.showMessageDialog(null,"Chamando a ficha prioritaria ..... " + ficha.getConteudo());        
                            filaPrioritaria.poll(); //faz a mesma coisa que o remove, so que se a fila estiver vazia, nao da erro
                        } else if (!filaNormal.isEmpty()) {
                            ficha = filaNormal.peek();                        
                            JOptionPane.showMessageDialog(null,"Chamando a ficha normal ..... " + ficha.getConteudo());        
                            filaNormal.poll(); //faz a mesma coisa que o remove, so que se a fila estiver vazia, nao da erro
                        }
                        break;                
                    default:
                        JOptionPane.showMessageDialog(null,"Opcao invalida");        
                        break;
                }
            } catch(Exception e) {
                break; //para sair do while
            }
        } while (!opcao.equals("4"));

        System.exit(0);

        
    }
}

File: 2023_manha/trabalhoFila_atendimento/atendimento.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#include "fila.h"

int main() {
    Fila filaNormal;
    Fila filaPrioritaria;
    int contadorFilaNormal = 1;
    int contadorFilaPrioritaria = 1000;
    int contadorAtendimentos = 0;
    inicializacao(&filaNormal);
    inicializacao(&filaPrioritaria);
    int opcao;
    Celula ficha;
    do {
        system("cls");
        printf("1 - Ficha normal\n");
        printf("2 - Ficha prioritaria\n");
        printf("3 - Chama ficha\n");
        printf("4 - Sair\n");
        printf("Opcao: ");
        scanf("%d", &opcao);

        switch (opcao) {
            case 1:
                //exibir a ficha ou imprimir a ficha
                printf("Ficha normal gerada: %d\n", contadorFilaNormal);
                inserirFila(contadorFilaNormal, &filaNormal);
                contadorFilaNormal++;
                break;
            case 2:
                //exibir a ficha ou imprimir a ficha
                printf("Ficha prioritaria gerada: %d\n", contadorFilaPrioritaria);
                inserirFila(contadorFilaPrioritaria, &filaPrioritaria);
                contadorFilaPrioritaria++;
                break;
            case 3:
                //definir qual fila atender
                if (contadorAtendimentos % 3 == 0) {
                    ficha = removerFila(&filaPrioritaria);
                    if (ficha.conteudo != -27) {
                        printf("Chamando ficha prioritaria: %d - %02d/%02d/%d %02d:%02d\n", ficha.conteudo, ficha.dataHora.tm_mday, ficha.dataHora.tm_mon + 1, ficha.dataHora.tm_year + 1900, ficha.dataHora.tm_hour, ficha.dataHora.tm_min);
                        contadorAtendimentos++;
                    } else {
                        if (filaNormal.cabeca != NULL) {
                            ficha = removerFila(&filaNormal);
                            printf("Chamando ficha normal: %d - %02d/%02d/%d %02d:%02d\n", ficha.conteudo, ficha.dataHora.tm_mday, ficha.dataHora.tm_mon + 1, ficha.dataHora.tm_year + 1900, ficha.dataHora.tm_hour, ficha.dataHora.tm_min);
                        }
                    }
                } else {
                    ficha = removerFila(&filaNormal);
                    if (ficha.conteudo != -27) {
                        printf("Chamando ficha normal: %d - %02d/%02d/%d %02d:%02d\n", ficha.conteudo, ficha.dataHora.tm_mday, ficha.dataHora.tm_mon + 1, ficha.dataHora.tm_year + 1900, ficha.dataHora.tm_hour, ficha.dataHora.tm_min);
                        contadorAtendimentos++;
                    } else {
                        if (filaPrioritaria.cabeca != NULL) {
                            ficha = removerFila(&filaPrioritaria);
                            printf("Chamando ficha prioritaria: %d - %02d/%02d/%d %02d:%02d\n", ficha.conteudo, ficha.dataHora.tm_mday, ficha.dataHora.tm_mon + 1, ficha.dataHora.tm_year + 1900, ficha.dataHora.tm_hour, ficha.dataHora.tm_min);
                        }
                    }
                }
                break;
            case 4:
                break;
            default:
                printf("Opcao invalida!\n");
                break;
        }
        system("pause");
    } while (opcao != 4);

    return 1;
}